web/source/osk/activeLayout.ts

```diff
@@ -11,22 +9,48 @@ namespace com.keyman.osk {
       pad: '15'
     };
 
+    id?: string;
+    width?: string;
+    pad?: string;
+    
+    proportionalX: number;
+    proportionalWidth: number;
+
+    static polyfill(key: LayoutKey) {
+      // Add class functions to the existing layout object, allowing it to act as an ActiveLayout.
+      let dummy = new ActiveKey();
+      for(let prop in dummy) {
+        if(!key.hasOwnProperty(prop)) {
+          key[prop] = dummy[prop];
+        }
+      }
+    }
+  }
+
+  class ActiveRow implements LayoutRow {
+    // Identify key labels (e.g. *Shift*) that require the special OSK font
+    static readonly SPECIAL_LABEL=/\*\w+\*/;
```

Comment:
Minor note about `/\w+/`. This only matches ASCII "word" characters; anything non-ASCII will not be matched, e.g.,:

```js
> /\w+/.test('hello')
true
> /\w+/.test('ᑖᓂᓯ')
false
```

I'm not sure if that is the intention in this case.

Reply:
That's fine in this case - this is only used for showing select key caps for keyboard layouts, such as the special symbol most of our keyboards show for Shift.  The `*____*` syntax tells KMW to replace the keycap text with the appropriate matching symbol, and that code must be specified with select ASCII codes.